ir/dict.go

```diff
@@ -25,6 +25,8 @@ func (p Pair) WritePretty(w io.Writer) error {
 	return nil
 }
 
+func (p Pair) encodeJSON() (interface{}, error) { return nil, nil }
```

Comment:
I had to add it because we were returning a `Pair` as a node in `dict.Remove`. I changed that function to return a boolean instead so we don't have to Pair as a node.